fuels-contract/src/contract.rs

```diff
@@ -91,6 +90,11 @@ impl Contract {
         #[allow(clippy::iter_cloned_collect)]
         let script = vec![
             Opcode::ADDI(0x10, REG_ZERO, script_data_offset),
+            // @todo currently there's no way to forward an amount.
+            // This would be done by programmatically changing
+            // $rB (`REG_ZERO` right now) to actually point to an
+            // amount. This would forward the amount in $rB using
+            // $rC as the asset_id.
```

Comment:
@adlerjohn Need your thoughts on this here. The issue being tackled here is this one https://github.com/FuelLabs/fuels-rs/issues/138. However, it only addresses the byte price and gas price/limit params. The original issue, on top of these params, also wants to be able to set amount/asset_id to be forwarded in the call (and be consumed with `msg_amount()` in Sway). 

But these aren't as simple as updating a value in the `Transaction` (like gas price, byte price, gas limit). I believe setting the amount/asset to forward means updating this assembly block, more specifically  `Opcode::CALL(0x10, REG_ZERO, 0x10, REG_CGAS)`, to actually be updated programmatically. Does that make sense? Is that the best/right approach?

Reply:
Your changes don't actually resolve #138 since that's about what gets forwarded in the call to the contract, which as you say here would require changes to the how it's called. You should unlink #138 from this PR since it's completely orthogonal.

Reply:
It's not completely orthogonal. The example I mentioned in https://github.com/FuelLabs/fuels-rs/issues/138 is orthogonal because it's covering only `value` (which is the value to be forwarded). But in the original discussion the motivation for that issue was to also programmatically set `gas_price`, `gas_limit`, and `byte_price`, which has been a blocker for @nfurfaro. 

At the beginning of this work, I tried to address it all (including `value`) but realized only `gas_price`, `gas_limit`, and `byte_price` could be done with this approach, whereas `value` (and `asset_id`) would require a totally different approach. Hence why I didn't address it here and am leaving it for another PR. I was also planning on creating another issue, extending https://github.com/FuelLabs/fuels-rs/issues/138.

Reply:
Updated https://github.com/FuelLabs/fuels-rs/issues/138 to limit its scope and created this one https://github.com/FuelLabs/fuels-rs/issues/159.